src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -424,15 +424,32 @@ void VideoDecoder::addVideoStreamDecoder(
   updateMetadataWithCodecContext(streamInfo.streamIndex, codecContext);
   streamInfo.options = options;
   int width = options.width.value_or(codecContext->width);
-  auto colorConversionLibrary = options.colorConversionLibrary.value_or(
-      ColorConversionLibrary::FILTERGRAPH);
-  bool useFilterGraph =
-      (colorConversionLibrary != ColorConversionLibrary::SWSCALE);
-  if (useFilterGraph) {
+
+  // Use swscale for color conversion by default because it is faster.
+  VideoDecoder::ColorConversionLibrary autoColorConversionLibrary =
```

Comment:
The actual logic of what we're doing here is pretty straight-forward, but I had to read it very carefully because of the length of the variable names their similarity to other locals. I think readability would be improved by pulling this into a function:
```
VideoDecoder::ColorConversionLibrary detectColorConversionLibrary(int width) {
  if (width % 32 != 0 && !canSwsScaleHandleUnalignedData()) {
    return ColorConversionLibrary::FILTERGRAPH;
  }
  return ColorConversionLibrary::SWSCALE;
}
```
And then locally:
```
auto colorConversionLibrary =
      options.colorConversionLibrary.value_or(detectColorConversionLibrary());
```